test/users/engin/compiler/nestedRecordField.future

```diff
@@ -0,0 +1,4 @@
+bug: inner record object as a field
+
+when outer is a record it seems to fail whereas when it is class it seems to be
+working fine
```

Comment:
Both are slightly different yes. One is an inner record dereferencing `outer`, the other is that a record having a field of an inner record type.

I discovered the latter while playing around with my test for the former.